src/scripts/ui/inline-roll-links.ts

```diff
@@ -348,9 +348,26 @@ export class InlineRollLinks {
             flags.pf2e.messageId = messageId;
         }
 
-        const actor = resolveActor(foundryDoc ?? resolveDocument(link));
-        if (actor || pf2Traits) {
+        const actor = resolveActor(foundryDoc);
+        const item =
+            foundryDoc instanceof ItemPF2e
+                ? foundryDoc
+                : foundryDoc instanceof ChatMessagePF2e
+                  ? foundryDoc.item
+                  : null;
+        if (actor || pf2Traits || item) {
             const origin: Record<string, unknown> = {};
+            if (item) {
```

Comment:
I double checked how origin is used elsewhere, including in strikes. item.getOriginData() is used instead of everything else if it exists. I would instead rewrite the entire origin assignment into an if/else, if there's an item, set flags.pf2e.origin to item.getOriginData()

Reply:
So to summarize, rewrite this to be `if (actor || pf2Traits) { ... } else if (item) { ... }` and get rid of stwlam's initial `uuid -> item` override?

Reply:
almost, but item first.

Reply:
right.